tests/components/sensor/test_fido.py

```diff
@@ -59,50 +59,60 @@ class PyFidoFakeModule():
-    config = {
-        'sensor': {
-            'platform': 'fido',
-            'name': 'fido',
-            'username': 'myusername',
-            'password': 'password',
-            'monitored_variables': [
-                'balance',
-                'data_remaining',
-            ],
+class TestSeason(unittest.TestCase):
+    """Test the season platform."""
+
+    fake_async_add_devices_called = False
+
+    def setUp(self):
+        """Setup things to be run when tests are started."""
+        self.hass = get_test_home_assistant()
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @asyncio.coroutine
+    def test_fido_sensor(self):
+        """Test the Fido number sensor."""
+        sys.modules['pyfido'] = MagicMock()
+        sys.modules['pyfido.client'] = MagicMock()
+        sys.modules['pyfido.client.PyFidoError'] = \
+            PyFidoErrorMock
+        import pyfido.client
+        pyfido.FidoClient = FidoClientMock
+        pyfido.client.PyFidoError = PyFidoErrorMock
+        config = {
+            'sensor': {
+                'platform': 'fido',
+                'name': 'fido',
+                'username': 'myusername',
+                'password': 'password',
+                'monitored_variables': [
+                    'balance',
+                    'data_remaining',
+                ],
+            }
         }
-    }
-    with assert_setup_component(1):
-        yield from async_setup_component(hass, 'sensor', config)
-    state = hass.states.get('sensor.fido_1112223344_balance')
-    assert state.state == "160.12"
-    assert state.attributes.get('number') == "1112223344"
-    state = hass.states.get('sensor.fido_1112223344_data_remaining')
-    assert state.state == "100.33"
-
-
-@asyncio.coroutine
-def test_error(hass, caplog):
-    """Test the Fido sensor errors."""
-    caplog.set_level(logging.ERROR)
-    sys.modules['pyfido'] = PyFidoFakeModule()
-    sys.modules['pyfido.client'] = PyFidoClientFakeModule()
-
-    config = {}
-    ret = yield from fido.async_setup_platform(hass, config,
-                                               fake_async_add_devices)
-    assert ret is None
+        with assert_setup_component(1):
+            yield from async_setup_component(self.hass, 'sensor', config)
+        state = self.hass.states.get('sensor.fido_1112223344_balance')
+        assert state.state == "160.12"
+        assert state.attributes.get('number') == "1112223344"
+        state = self.hass.states.get('sensor.fido_1112223344_data_remaining')
+        assert state.state == "100.33"
+
+    @asyncio.coroutine
+    def test_error(self):
+        """Test the Fido sensor errors."""
+        sys.modules['pyfido'] = PyFidoFakeModule()
+        sys.modules['pyfido.client'] = PyFidoClientFakeModule()
+
+        config = {}
+        yield from fido.async_setup_platform(self.hass, config,
+                                             self.fake_async_add_devices)
```

Comment:
You don't need to change everything like that. Just pass in a mock instead of `async_add_devices`. You can easily check if the mock was called via the called attribute. There's a lot of conveniency in using mocks overall.

Reply:
Wow thanks ! I'm an expert with mock/unittest !